pkg/api/types.go

```diff
@@ -256,6 +258,19 @@ type SecretVolumeSource struct {
 	Target ObjectReference `json:"target"`
 }
 
+// NFS represents and NFS Mount that lasts the lifetime of a pod
+type NFSVolumeSource struct {
+	// Server is the hostname or ip address of the NFS server
+	Server string `json:"server"`
+
+	// SourcePath is the exported NFS share
+	SourcePath string `json:"sourcePath"`
+
+	// MountOptions is a comma delimited list of mount options. The available options are listed on the mount man page as arguments to -o.
+	// As an example: noatime,nodev,nosuid,ro
+	MountOptions string `json:"mountOptions"`
```

Comment:
readonly will be common across many implementations. It would be nice for it to be supported uniformly.

In general, I believe in providing abstraction, where it provides value, and won't get in the way.

Would a cluster/storage administrator be specifying the mount options, or the end user?

Does this need to be the primary configuration mechanism, or just an escape hatch? Is there a common set (or sets) of options that would be used most of the time?

Could a machine be bricked by bad options?  If so, then we could permit this, but we'd add it to our list of "protected" features, such as host ports, host directories, host networking (if we add that), privileged mode, capabilities, etc.

What happens if an invalid list of options were specified? What's going to be checking them? How would that be reported back to the user?